src/sage/arith/long.pxd

```diff
@@ -310,6 +321,9 @@ cdef inline bint integer_check_long_py(x, long* value, int* err):
     cdef long lead
     cdef long lead_2_overflow = (<long>1) << (BITS_IN_LONG - PyLong_SHIFT)
     cdef long lead_3_overflow = (<long>1) << (BITS_IN_LONG - 2 * PyLong_SHIFT)
+    if BITS_IN_LONG < 2 * PyLong_SHIFT:
```

Comment:
Would it not be better to check this before executing the UB?

Reply:
On second thought, I think you are right, since UB could eat the thermal paste in our cpu, we better make sure we don't execute it at all, so I did.